ci/docker_cache.py

```diff
@@ -52,7 +53,7 @@ def build_save_containers(platforms, registry, load_cache) -> int:
     if len(platforms) == 0:
         return 0
 
-    platform_results = Parallel(n_jobs=len(platforms), backend="multiprocessing")(
+    platform_results = Parallel(n_jobs=PARALLEL_BUILDS, backend="multiprocessing")(
```

Comment:
Will this cause the problems for memory usage? As we tested, if three Scala Integration test running at the sametime would make the CI super slow

Reply:
This only builds the containers and doesn't run tests - the memory footprint is quite small

Reply:
Thus, please change it back to len(platforms) since this will otherwise slow down the docker cache generation process.

Reply:
No, it's too many jobs in parallel out of control. 10 is a reasonable number and there's no speed constraints. Stop inventing performance problems that don't exist.

Reply:
@lanking520 this is not running tests, just building containers. I don't get your concern.